File: leetcode.com/0814_binary_tree_pruning/solution.c
struct TreeNode *pruneTree(struct TreeNode *root)
{
    if (root->left && pruneTree(root->left) == NULL)
        root->left = NULL;

    if (root->right && pruneTree(root->right) == NULL)
        root->right = NULL;

    if (root->val == 0 && root->left == NULL && root->right == NULL)
        return NULL;

    return root;
}

File: leetcode.com/0814_binary_tree_pruning/solution.go
func pruneTree(root *TreeNode) *TreeNode {
    if root.Left != nil && pruneTree(root.Left) == nil {
        root.Left = nil
    }

    if root.Right != nil && pruneTree(root.Right) == nil {
        root.Right = nil
    }

    if root.Val == 0 && root.Left == nil && root.Right == nil {
        return nil
    }

    return root;
}
